### PR TITLE
[TRUNK-5299] Possible NPE due to use of Objects.requireNonNull()

### PR DESCRIPTION
**Pull request by BartlomiejRasztabiga**
Issue: https://issues.openmrs.org/browse/TRUNK-5299

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/ModuleFactory.java [hunk 1]
```diff
@@ -171,9 +171,10 @@ public static void loadModules() {
...
 		}
-		
-		if (modulesFolder.isDirectory()) {
-			loadModules(Arrays.asList(Objects.requireNonNull(modulesFolder.listFiles())));
+
+		File[] fileList = modulesFolder.listFiles();
+		if (modulesFolder.isDirectory() && fileList != null) {
+			loadModules(Arrays.asList(fileList));
 		} else {
 			log.error("modules folder: '" + modulesFolder.getAbsolutePath() + "' is not a valid directory");
```
Comment (teleivo) [inline api/src/main/java/org/openmrs/module/ModuleFactory.java]: now the log message also includes a directory which are empty which I assume is still a valid directory.

Comment (BartlomiejRasztabiga) [reply api/src/main/java/org/openmrs/module/ModuleFactory.java]: moduleFolder.listFiles() returns null only when directory is not valid

**Comment by teleivo** [comment]
closing this since we should reopen the issues that introduced these changes and fix them

**Comment by BartlomiejRasztabiga** [comment]
Okay, should I reopen them or you will do it and ping me after?

**Pull request closed without merging**